### ISSUE TITLE
kpasswd silently fails when new passwords do not agree

### ISSUE THREAD

**Issue by macrotex** [heimdal/heimdal#616]
This affects version Heimdal 7.7.0 (and earlier versions, as well).

Here is an attempt to change password using `kpasswd` where the two new password entries are the _same_:

```
# kpasswd johndoe
johndoe@example.com's Password: 
New password for johndoe@example.com: 
Verify password - New password for johndoe@example.com: 
Success : Password changed
# echo $?
0
```

Here is the what you see when the two new password entries are _different_:

```
# kpasswd johndoe
johndoe@example.com's Password: 
New password for johndoe@example.com: 
Verify password - New password for johndoe@example.com: 
# echo $?
1
```

What I would like to see is some sort of error message when the two new password entries are different, e.g., "Password change failed: the new passwords did not agree".

**Comment by lhoward** [comment]
This was fixed in 014f16883c7c911b2804fb081e885e0571528e85. @jaltman, can we close?

**Comment by jaltman** [comment]
@lhoward Can 014f16883c7c911b2804fb081e885e0571528e85 be easily cherry-picked to heimdal-7-1-branch?
